include/zephyr/kernel.h

```diff
@@ -214,24 +212,18 @@ void k_thread_foreach_unlocked(
  */
 #define K_CALLBACK_STATE (BIT(4))
 
-#ifdef CONFIG_ARC
-/* ARC processor Bitmask definitions for threads user options */
-
-#if defined(CONFIG_ARC_DSP_SHARING)
```

Comment:
The #if allowed to allocate the scarce option bits differently for different architectures/configured features, but as all options still fit and there is no overlap I guess the #ifdef isn't required.
If needed, the #ifdef can always be added back later.